contrib/twitchtv/twirp/twirp.go

```diff
@@ -167,16 +169,23 @@ func requestReceivedHook(cfg *config) func(context.Context) (context.Context, er
 		if !math.IsNaN(cfg.analyticsRate) {
 			opts = append(opts, tracer.Tag(ext.EventSampleRate, cfg.analyticsRate))
 		}
-		_, ctx = tracer.StartSpanFromContext(ctx, spanNameFromContext(ctx), opts...)
+		span, ctx := tracer.StartSpanFromContext(ctx, spanNameFromContext(ctx), opts...)
+
+		ctx = context.WithValue(ctx, twirpSpanKey, span)
+		
 		return ctx, nil
 	}
 }
 
 func requestRoutedHook(cfg *config) func(context.Context) (context.Context, error) {
 	return func(ctx context.Context) (context.Context, error) {
-		span, ok := tracer.SpanFromContext(ctx)
+		maybeSpan := ctx.Value(twirpSpanKey)
+		if maybeSpan == nil {
+			return ctx, errors.New("no span in context")
```

Comment:
We (almost) never want tracer errors to bubble up into the user's application to avoid causing application failures. Instead of returning the error, we want to handle the error by logging it. Something like:
```
log.Error("contrib/twitchtv/twirp.requestRoutedHook: found no span in context")
```

Reply:
Done.